src/firefly/java/edu/caltech/ipac/firefly/server/servlets/HealthCheck.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+package edu.caltech.ipac.firefly.server.servlets;
+
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.util.FileUtil;
+import edu.caltech.ipac.util.StringUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Check application health by performing a simple task, then return HTTP code 200 if succeed.
+ *
+ * optional parameters:
+ *  mem=<size-in-mb>    : amount of memory to allocate, between 1-1024 MB; default to 5
+ *
+ * Date: Oct 12, 2020
+ *
+ * @author loi
+ * @version $Id: ServerStatus.java,v 1.1 2009/06/04 00:12:42 loi Exp $
+ */
+public class HealthCheck extends BaseHttpServlet {
+
+    protected void processRequest(HttpServletRequest req, HttpServletResponse res) throws Exception {
+
+        // memory check
+        int mem = Math.max(Math.min(StringUtils.getInt(req.getParameter("mem"), 5), 1024), 1);
+        byte[] b = new byte[mem * 1024 * 1024];
+        Runtime rt= Runtime.getRuntime();
+        Logger.debug(String.format("HealthCheck(Used/Max/Allocated): %s/%s/%s",
+                            FileUtil.getSizeAsString(rt.totalMemory() - rt.freeMemory()),
+                            FileUtil.getSizeAsString(rt.maxMemory()), FileUtil.getSizeAsString(b.length)));
+    }
```

Comment:
You might want to catch an `OutOfMemoryError` here. and log that as well.

Reply:
I am guessing this is called outside of firefly app context. I would think it should return a string such a `Health Check Passed` or something.

You could also return a `Health Check Failed` for an `OutOfMemoryError` but I doubt it would return in the case. I have enchanted  edge case when we do get an`OutOfMemoryError` right before the server hangs.

Reply:
This is not meant for interactive use.  It's an HTTP API where a 200 status is pass and everything else is fail.
In the case of OOME or any exception, it will be caught upstream and a 500 will be returned.
This is usually used with a `timeout` threshold.  When it failed to return within a specified time, it is also considered 'unhealthy' or fail.

The `/healthz` check is within an app context.  So, for firefy, it would be `/firefly/healthz`.

Reply:
If you are logging it is still worth catching the `OOME`

Reply:
I can if you like, for logging purposes.  But in most cases, you won't even get an exception.  It will just hang there.

Normally, I don't want to log this.  It's unnecessary noise.  But, `debug` is our lowest log level.  This segue into our need to move most of our logs to 'info' level and set that as the default log level.

Reply:
the OOME is catch anyway in catalina.out, this is redundant to add it here in my opinion.

Reply:
I also think it's redundant, since the code will look like this.  However, I am fine with adding it. 
```
try { 
... 
} catch(oome) {
  log(oome);
  throw oome;
}
```